cocotb/share/makefiles/Makefile.inc

```diff
@@ -36,8 +36,7 @@ COCOTB_PY_DIR := $(realpath $(shell cocotb-config --prefix))
 # simulations: Makefile fragments, and the simulator libraries.
 COCOTB_SHARE_DIR := $(COCOTB_PY_DIR)/cocotb/share
 
-ARCH?=$(shell uname -m)
-export ARCH
+PYTHON_ARCH := $(shell $(PYTHON_BIN) -c 'from platform import architecture; print(architecture()[0])')
```

Comment:
Just as a safety net for the assumptions in all other code, please check for 32bit and 64bit and bail out if it's anything else.